app/repositories/DepartureRepository.scala

```diff
@@ -518,6 +522,30 @@ class DepartureRepository @Inject()(
         }
     }
   }
+
+  def deleteCompletedJourneys(): Future[Try[Unit]] = {
+    val writeOffSelector = Json.obj("messages.messageType" -> WriteOffNotification.code)
+    val rejectedSelector = Json.obj("messages.messageType" -> DeclarationRejected.code)
+    val messageSelector  = Json.obj("$or"                  -> Json.arr(writeOffSelector, rejectedSelector))
+
+    val instant          = LocalDateTime.now(clock)
+    val midnight         = LocalDateTime.of(instant.getYear, instant.getMonth, instant.getDayOfMonth, 0, 0, 0)
```

Comment:
There is a `LocalDate` method `atStartOfDay` which will give you a `ZonedDateTime` - that's probably better for doing this kind of deletion:
```suggestion
    val instant  = LocalDate.now(clock)
    val midnight = instant.atStartOfDay(ZoneOffset.UTC)
```

Reply:
In one of the ATaR projects we actually had an even more tricky requirement where we had to show rulings that were active up to midnight in the current UK timezone - I'm not sure what we will want to do in this project but it ended up looking something like this:
```scala
    val startOfToday = LocalDate.now(clock).atStartOfDay // Get a LocalDateTime at midnight
    val zoneOffset   = ZoneId.of("Europe/London").getRules.getOffset(startOfToday) // Get the timezone offset on that date in the UK 
    val today        = startOfToday.toInstant(zoneOffset) // Now we can finally create the ZonedDateTime that represents that timestamp
```